### PR TITLE
MainMenu: Create function to deal with user input of startPoint and L…

### LINKED ISSUE #477: length can't go over 30m

### PR DESCRIPTION
**Pull request by patrickelectric**
…ength

Fix #477

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: qml/MainMenu.qml [hunk 1]
```diff
@@ -166,20 +166,64 @@ Item {
...
+                    property var minLengthMm: 1e3
+
+                    function setDepth(startMm, lengthMm) {
+                        // Sensor max range is limited to maxLengthMm, this value is also the max window depth
+                        // The minimum range between start and length is minLengthMm
+
+                        // Start
+                        if(ping.start_mm !== startMm) {
+                            // Check if length + new start is bigger than our limit (maxLengthMm)
+                            if(startMm >= maxLengthMm - minLengthMm) {
```
Comment (jaxxzer) [inline qml/MainMenu.qml]: Should you be using `maxDepthMm` where you are using `maxLengthMm`?

File: qml/MainMenu.qml [hunk 2]
```diff
@@ -166,20 +166,64 @@ Item {
...
+                                startMm = maxLengthMm - minLengthMm
+                            }
+
+                            // Limit window to maxLengthMm
+                            if(startMm + ping.length_mm > maxLengthMm) {
```
Comment (jaxxzer) [inline qml/MainMenu.qml]:
```suggestion
                            if(startMm + ping.length_mm > maxDepthMm) {
```

File: qml/MainMenu.qml [hunk 3]
```diff
@@ -166,20 +166,64 @@ Item {
...
+
+
                     PingTextField {
                         id: startLength
                         title: "Scan start point (mm):"
                         text: ping.start_mm
                         validator: IntValidator {
                             bottom: 0
-                            top: SettingsManager.debugMode ? 1e6 : 7e5
+                            top: maxDepthMm
```
Comment (jaxxzer) [inline qml/MainMenu.qml]: `[12:52:22.105] default[Warning]: MainMenu.qml(221) qrc:/MainMenu.qml:221: ReferenceError: maxDepthMm is not defined`

File: qml/MainMenu.qml (hunk 1 above)
Comment (patrickelectric) [reply qml/MainMenu.qml]: maxDepthMm is the correct variable, something was lost while squashing

**Pull request merged**